src/Umbraco.Infrastructure/Migrations/Upgrade/V_8_0_0/DataTypes/PreValueMigratorComposer.cs

```diff
@@ -2,8 +2,7 @@
 
 namespace Umbraco.Core.Migrations.Upgrade.V_8_0_0.DataTypes
 {
-[RuntimeLevel(MinLevel = RuntimeLevel.Upgrade, MaxLevel = RuntimeLevel.Upgrade)] // only on upgrades
-public class PreValueMigratorComposer : ICoreComposer
+    public class PreValueMigratorComposer : ICoreComposer
```

Comment:
What ensures this only runs on upgrades now?

Reply:
Composers add services, that should always happen.

Components etc use those services, migration story should involve checking runtime state just not at registration time.

All the discussion in #8653 is currently on this topic.